lang/cs/Org.Apache.REEF.Driver/Bridge/Events/FailedEvaluator.cs

```diff
@@ -30,35 +31,50 @@
 namespace Org.Apache.REEF.Driver.Bridge.Events
 {
     [DataContract]
-    internal class FailedEvaluator : IFailedEvaluator
+    internal sealed class FailedEvaluator : IFailedEvaluator
     {
         private static readonly Logger LOGGER = Logger.GetLogger(typeof(FailedEvaluator));
+        private readonly string _id;
 
         public FailedEvaluator(IFailedEvaluatorClr2Java clr2Java)
         {
             InstanceId = Guid.NewGuid().ToString("N");
             FailedEvaluatorClr2Java = clr2Java;
             EvaluatorRequestorClr2Java = FailedEvaluatorClr2Java.GetEvaluatorRequestor();
-            Id = FailedEvaluatorClr2Java.GetId();
+            _id = FailedEvaluatorClr2Java.GetId();
         }
 
         [DataMember]
         public string InstanceId { get; set; }
 
-        public string Id { get; private set; }
-
-        public EvaluatorException EvaluatorException { get; set; }
-
-        public List<FailedContext> FailedContexts { get; set; }
-
-        public Optional<IFailedTask> FailedTask { get; set; }
-
         [DataMember]
         private IFailedEvaluatorClr2Java FailedEvaluatorClr2Java { get; set; }
 
         [DataMember]
         private IEvaluatorRequestorClr2Java EvaluatorRequestorClr2Java { get; set; }
 
+        public string Id
+        {
+            get { return _id; }
+        }
+
+        //TODO[REEF-769]: Implement
+        public EvaluatorException EvaluatorException
+        {
+            get { return null; }
+        }
+
+        //TODO[REEF-769]: Implement
+        public IList<IFailedContext> FailedContexts
+        {
+            get { return new List<IFailedContext>(0); }
+        }
+        //TODO[REEF-769]: Implement
+        public Optional<IFailedTask> FailedTask
+        {
+            get { return Optional<IFailedTask>.Empty(); }
+        }
+
         public IEvaluatorRequestor GetEvaluatorRequetor()
```

Comment:
Typo. Are we allowing breaking changes now? If so could you fix it? If not deprecate it and return from the same private function. May also be better to follow convention with ``EvaluatorRequestor { get; }``